[Gen-AI] Sync indexing tools breaks as PGVector is async

I noticed that PGVector is instanced in async mode here : https://github.com/theopenconversationkit/tock/blob/master/gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/vector_stores/pgvector_factory.py#L53 

Which lead to the following error when using index_documents.py (toolings) this is now sync (not async) : 
```bash
INFO:index_documents:i=0, splitted_docs=6
Traceback (most recent call last):
  File "tock/gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/index_documents.py", line 426, in <module>
    details = index_documents()
  File "tock/gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/index_documents.py", line 183, in index_documents
    embedding_and_indexing(splitted_docs, vector_store)
  File "tock/gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/index_documents.py", line 273, in embedding_and_indexing
    vector_store.add_documents(documents=splitted_docs[i: i + embedding_bulk_size], bulk_size=embedding_bulk_size)
  File "tock/gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/.venv/lib/python3.10/site-packages/langchain_core/vectorstores/base.py", line 287, in add_documents
    return self.add_texts(texts, metadatas, **kwargs)
  File "tock/gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/.venv/lib/python3.10/site-packages/langchain_postgres/vectorstores.py", line 883, in add_texts
    assert not self._async_engine, "This method must be called without async_mode"
AssertionError: This method must be called without async_mode
```

One way to solve it would be to let the use choose the method used sync / async optionally in the settings, enforce it as async when the factory is used in the indexing tools maybe.

--- a/gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/vector_stores/open_search_factory.py
+++ b/gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/vector_stores/open_search_factory.py
@@ -14,6 +14,7 @@
 #
 """Model for creating OpenSearchFactory"""
 import logging
+from typing import Optional
 
 from langchain_community.vectorstores.opensearch_vector_search import OpenSearchVectorSearch
 from langchain_core.vectorstores import VectorStoreRetriever
@@ -41,7 +42,7 @@ class OpenSearchFactory(LangChainVectorStoreFactory):
 
     setting: OpenSearchVectorStoreSetting
 
-    def get_vector_store(self) -> OpenSearchVectorSearch:
+    def get_vector_store(self, async_mode: Optional[bool] = True) -> OpenSearchVectorSearch:
         password = fetch_secret_key_value(self.setting.password)
         logger.info(
             'OpenSearch user credentials: %s:%s',

--- a/gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/vector_stores/pgvector_factory.py
+++ b/gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/vector_stores/pgvector_factory.py
@@ -14,6 +14,7 @@
 #
 """Model for creating PGVectorFactory"""
 import logging
+from typing import Optional
 
 from langchain_core.vectorstores import VectorStoreRetriever
 from langchain_postgres import PGVector
@@ -37,7 +38,7 @@ class PGVectorFactory(LangChainVectorStoreFactory):
 
     setting: PGVectorStoreSetting
 
-    def get_vector_store(self) -> PGVector:
+    def get_vector_store(self, async_mode: Optional[bool] = True) -> PGVector:
         password = fetch_secret_key_value(self.setting.password)
         logger.info(
             'PostgreSQL user credentials: %s:%s',
@@ -50,7 +51,7 @@ class PGVectorFactory(LangChainVectorStoreFactory):
             collection_name=self.index_name,
             connection=f'postgresql+psycopg://{self.setting.username}:{password}@{self.setting.host}:{self.setting.port}/{self.setting.database}',
             use_jsonb=True,
-            async_mode=True
+            async_mode=async_mode
         )
 
     def get_vector_store_retriever(self, search_kwargs: dict) -> VectorStoreRetriever:

--- a/gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/vector_stores/vector_store_factory.py
+++ b/gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/vector_stores/vector_store_factory.py
@@ -16,7 +16,7 @@
 
 import logging
 from abc import ABC, abstractmethod
-from typing import List
+from typing import List, Optional
 
 from langchain_core.documents import Document
 from langchain_core.embeddings import Embeddings
@@ -42,9 +42,11 @@ class LangChainVectorStoreFactory(ABC, BaseModel):
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
     @abstractmethod
-    def get_vector_store(self) -> VectorStore:
+    def get_vector_store(self, async_mode: Optional[bool] = True) -> VectorStore:
         """
         Fabric the Vector Store.
+        Args:
+            async_mode: enable/disable the async_mode for vector DB client (if supported). Default to True.
         :return: VectorStore the interface for Vector Database.
         """
         pass

--- a/gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/index_documents.py
+++ b/gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/index_documents.py
@@ -178,7 +178,7 @@ def index_documents() -> IndexingDetails:
         embedding_function=embeddings
     )
     vector_store_factory.check_vector_store_connection()
-    vector_store = vector_store_factory.get_vector_store()
+    vector_store = vector_store_factory.get_vector_store(async_mode=False)
 
     embedding_and_indexing(splitted_docs, vector_store)
 

